### ISSUE TITLE
add widest (/maximum capacity) path algorithm

### ISSUE THREAD

**Issue by zurKreuzigungLinks** [JuliaGraphs/LightGraphs.jl#1557]
"In graph algorithms, the widest path problem is the problem of finding a path between two designated vertices in a weighted graph, maximizing the weight of the minimum-weight edge in the path. The widest path problem is also known as the maximum capacity path problem."
[https://en.wikipedia.org/wiki/Widest_path_problem](url)

I needed this and implemented it as it is quite simple. I just duplicated your functions and changed them a bit like so:
1. Use existing shortest path algorithm
2. Change edge weights w to 1/w: `distmx::AbstractMatrix{T}=1 ./ weights(g)`
3. Use the max edge weight as new distance:`tentative_g_score = max(g_score[current] + distmx[current, neighbor])`
4. Change Priority to the current "distance":`priority = tentative_g_score`

That should be it if I didn't forget anything.

**Comment by saolof** [comment]
Right, the widest path problem is fairly straightforwardly reduced to the shortest/longest (non-cyclic) path problem. You replace addition with max-addition basically.

**Comment by sbromberger** [comment]
This might be better in `LightGraphsFlows.jl` where they are concerned with capacity algorithms.
